meetings/2024-04/april-08.md

```diff
@@ -0,0 +1,1049 @@
+
+CDA: I think you did a good job. I just had these slides from a previous meeting. Yes, I probably should update this data. They’re approved at the GA meetings or reviewed and potentially approved at the GA meeting. So let us know if you have anyone good in mind. And it’s helpful to have, like, the nomination, like, text itself, but don’t let that stop you from, you know, providing your idea and we can help you with that as well. Thank you.
+
+### Speaker's Summary of Key Points
+
+- Reminder to take action on Ecma Recognition Awards
+
+## ECMA262 Status Updates
+
+Presenter: Kevin Gibbons (KG)
+
+- [slides](https://docs.google.com/presentation/d/1mXvZ-hgyjFca8BzqZ5jzFSa3AdCcr3rFkINaeZ857nA/edit)
+
+KG: Okay. Good morning, everyone. Or good whatever it is. This will be your typical brief update from the 262 editors. We’ve landed a handful of normative changes. I believe so the first three of these were decided in the previous meeting. This last item was -- in the process of tweaking some of the, I believe this was related to the first item where we are tweaking the semantics for CSP for foul. We accidentally made a normative change that was not intended and did not have consensus, and that no one caught during the review process, so we made another normative change to put it back. We don’t ask for consensus for bug fixes like that because they’re just restoring the -- what was already consensus, but we’d like to call them out anyway in case anyone was confused by the previous state or seeing that change go through.
+
+KG: Not much in the way of editorial changes to call out. The first one here, we’re calling out really only as an example of sort of a design principle. Which is that there is an abstract operation for finding the index of one string within another. Basically string.prototype.index of. And we tweaked it so that instead of returning negative one when the value wasn’t found, it returns a special value, and this is the approach that the editors intend to take for operations like this going forward, and if you’re writing a proposal and designing such an abstract operation yourself, we recommend this approach to you as well. It just generally makes it easier for readers and since the spec is not code, it’s not like there’s any overhead associated with returning a special value instead of returning negative 1. Not returning a number doesn’t actually affect anything. This isn’t C. So just something to keep in mind when designing things like this in the future.
+
+KG: Also, quite a few miscellaneous use cleanups and consistency fixes that we don’t feel are worth calling to the committee’s attention, but of curious you’re welcome ore view the commit history on GitHub. Not going to go through the list of upcoming work, because it hasn’t changed. We are still plugging away at a bunch of stuff. Most of these things we’re not actively working on. Some of them we are.
```

Comment:
```suggestion
KG: Also, quite a few miscellaneous use cleanups and consistency fixes that we don’t feel are worth calling to the committee’s attention, but if curious you’re welcome to view the commit history on GitHub. Not going to go through the list of upcoming work, because it hasn’t changed. We are still plugging away at a bunch of stuff. Most of these things we’re not actively working on. Some of them we are.
```